homeassistant/components/xiaomi_miio/select.py

```diff
@@ -0,0 +1,208 @@
+from enum import Enum
+import logging
+
+from miio.airhumidifier import LedBrightness as AirhumidifierLedBrightness
+from miio.airhumidifier_miot import LedBrightness as AirhumidifierMiotLedBrightness
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.core import callback
+
+from .const import (
+    CONF_DEVICE,
+    CONF_FLOW_TYPE,
+    CONF_MODEL,
+    DOMAIN,
+    FEATURE_FLAGS_AIRHUMIDIFIER,
+    FEATURE_FLAGS_AIRHUMIDIFIER_CA4,
+    FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB,
+    FEATURE_SET_LED_BRIGHTNESS,
+    KEY_COORDINATOR,
+    KEY_DEVICE,
+    MODEL_AIRHUMIDIFIER_CA1,
+    MODEL_AIRHUMIDIFIER_CA4,
+    MODEL_AIRHUMIDIFIER_CB1,
+    MODELS_HUMIDIFIER,
+    SERVICE_SET_LED_BRIGHTNESS,
+)
+from .device import XiaomiCoordinatedMiioEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_LED_BRIGHTNESS = "led_brightness"
+
+SERVICE_TO_METHOD = {
+    SERVICE_SET_LED_BRIGHTNESS: {
+        "method": "async_set_led_brightness",
+        "property": ATTR_LED_BRIGHTNESS,
+    },
+}
+
+
+@dataclass
+class SelectorType:
+    """Class that holds device specific info for a xiaomi aqara or humidifier selectors."""
+
+    name: str = None
+    icon: str = None
+    short_name: str = None
+    options: list = None
+    service: str = None
+
+
+SELECTOR_TYPES = {
+    FEATURE_SET_LED_BRIGHTNESS: SelectorType(
+        name="Led brightness",
+        icon="mdi:brightness-6",
+        short_name=ATTR_LED_BRIGHTNESS,
+        options=["Bright", "Dim", "Off"],
+        service=SERVICE_SET_LED_BRIGHTNESS,
+    ),
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Selectors from a config entry."""
+    entities = []
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        model = config_entry.data[CONF_MODEL]
+        device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+        coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+        device_features = 0
+        entity_class = None
+
+        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+
+        if model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
+            entity_class = XiaomiAirHumidifierSelector
+        elif model in [MODEL_AIRHUMIDIFIER_CA4]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+            entity_class = XiaomiAirHumidifierMiotSelector
+        elif model in MODELS_HUMIDIFIER:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER
+            entity_class = XiaomiAirHumidifierSelector
+        else:
+            return
+
+        for feature in SELECTOR_TYPES:
+            selector = SELECTOR_TYPES[feature]
+            if feature & device_features and feature in SELECTOR_TYPES:
```

Comment:
Do we even need this check? Don't we already know that the models above support an entity?

Reply:
Removed check and evaluation of device_features for now